File: app/services/auth_service.py
from sqlalchemy.orm import Session
from app.core.security import verify_password
from app.models.usuario import Usuario

class AuthService:
    @staticmethod
    def authenticate(db: Session, email: str, password: str) -> Usuario | None:
        user = db.query(Usuario).filter(Usuario.email == email).first()
        if not user or not verify_password(password, user.hashed_password):
            return None
        return user

auth_service = AuthService()

File: src/services/auth.service.ts
import { pool } from '../db/connection';
import { verifyPassword } from '../utils/password';
import { RowDataPacket } from 'mysql2';

export const authenticate = async (email: string, password: string) => {
  const [rows] = await pool.query<RowDataPacket[]>(
    `SELECT u.*, r.nombre as rol_nombre FROM usuario u
     JOIN rol r ON u.rol_id = r.id
     WHERE u.email = ?`,
    [email]
  );
  if (rows.length === 0) return null;
  const user = rows[0];
  const valid = await verifyPassword(password, user.hashed_password);
  if (!valid) return null;
  return user;
};
